### PR TITLE
ALFREDAPI-428 Change getPerson to getPersonOrNull in PeopleService

### PR DESCRIPTION
**Pull request by Ranecdev**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-428

- [X] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [X] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [X] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](https://github.com/xenit-eu/alfred-api/blob/master/developer-documentation)?
- [X] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [X] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

File: apix-impl/src/main/java/eu/xenit/apix/alfresco/people/PeopleService.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ private AuthorityType AuthorityTypeToAlfresco(eu.xenit.apix.people.AuthorityType
 
     @Override
     public Person GetPerson(String userName) {
-        return GetPerson(c.apix(alfrescoPersonService.getPerson(normalizeUserName(userName))));
+        return GetPerson(c.apix(alfrescoPersonService.getPersonOrNull(normalizeUserName(userName))));
```
Comment (vierbergenlars) [inline apix-impl/src/main/java/eu/xenit/apix/alfresco/people/PeopleService.java:130]: I think this should throw an exception when a person does not exist.

**Comment by Ranecdev** [comment]
> In addition to my comment, the REST API layer should return a 404 response when a non-existing person is requested.

The 404 response was already inplace for getPerson(userName) see PeopleWebscript#line72. I also added it for getPerson(nodeRef). Or should I create a new ticket for this

File: apix-rest-v1/src/main/java/eu/xenit/apix/rest/v2/people/PeopleWebscript.java [hunk 2]
```diff
@@ -43,6 +43,11 @@ public void getPerson(@UriVariable final String space, @UriVariable final String
             @UriVariable final String guid, WebScriptResponse webScriptResponse) throws IOException {
         logger.debug("Asked person with guid: " + guid);
         Person p = personService.GetPerson(createNodeRef(space, store, guid));
+        if (p == null) {
+            webScriptResponse.setStatus(404);
+            webScriptResponse.getWriter().write("Person does not exist");
```
Comment (vierbergenlars) [inline apix-rest-v1/src/main/java/eu/xenit/apix/rest/v2/people/PeopleWebscript.java:48]: Hmm, returning a non-JSON body as an error response in a JSON API?

Either returning no body (404 already implies that the requested resource doesn't exist), or a JSON object would be fine, I think

Comment (Ranecdev) [reply apix-rest-v1/src/main/java/eu/xenit/apix/rest/v2/people/PeopleWebscript.java:49]: Ok. I just copied the code over. Now added a contentType.

**Pull request merged**